databricks_cli/unity_catalog/cred_cli.py

```diff
@@ -33,9 +33,83 @@
+            'access_connector_id': az_mi_access_connector_id,
+            'managed_identity_id': az_mi_id
+        }
+
+    if ((gcp_sak_email is not None) or (gcp_sak_private_key_id is not None) or 
+        (gcp_sak_private_key is not None)):
+        data['gcp_service_account_key'] = {
+            'email': gcp_sak_email,
+            'private_key_id': gcp_sak_private_key_id,
+            'private_key': gcp_sak_private_key
+        }
+
 
 @click.command(context_settings=CONTEXT_SETTINGS,
                short_help='Create storage credential.')
+@click.option('--name', default=None,
+              help='Name of new storage credential')
+@click.option('--aws-iam-role-arn', default=None,
+              help='The Amazon Resource Name (ARN) of the AWS IAM role for S3 data access.')
+@click.option('--az-sp-directory-id', default=None,
+              help=(
+                'The directory ID corresponding to the Azure Active Directory (AAD) '
+                'tenant of the application.'))
+@click.option('--az-sp-application-id', default=None,
+              help=(
+                'The application ID of the application registration within the referenced '
+                'AAD tenant.'))
+@click.option('--az-sp-client-secret', default=None,
+              help='The client secret generated for the above app ID in AAD.')
+@click.option('--az-mi-access-connector-id', default=None,
+              help=(
+                'The Azure resource ID of the Azure Databricks Access Connector. Use the format, '
+                '/subscriptions/{guid}/resourceGroups/{rg-name}/providers/Microsoft.Databricks'
+                '/accessConnectors/{connector-name} .'))
+@click.option('--az-mi-id', default=None,
+              help=(
+                'The Azure resource ID of the managed identity. Use the format, '
+                '/subscriptions/{guid}/resourceGroups/{rg-name}/providers'
+                '/Microsoft.ManagedIdentity/userAssignedIdentities/{identity-name} .'
+                'This is only available for user-assigned identities. '
+                'For system-assigned identities, access-connector-id is used to identify '
+                'the identity. '
+                'If this flag is not provided, then we assume that it is using the system-assigned '
+                'identity.'))
+@click.option('--gcp-sak-email', default=None,
+              help=(
+                'Credential for GCP Service Account Key. '
+                'The email of the service account.'))
+@click.option('--gcp-sak-private-key-id', default=None,
+              help=(
+                'Credential for GCP Service Account Key. '
+                'The ID of the service account\'s private key.'))
+@click.option('--gcp-sak-private-key', default=None,
+              help=(
+                'Credential for GCP Service Account Key. '
+                'The service account\'s RSA private key.'))
+@click.option('--comment', default=None,
+              help='Free-form text description.')
```

Comment:
Can you reuse these options between create and update by any chance?

Impossible to review for completeness if they are separate.

Reply:
Grouped common decorators together.